CPP/SortingAlgrothims/Sort.cpp

```diff
@@ -73,34 +75,36 @@ void SortingAlgrothim<T>::InsertionSort(T arr[], uint32_t Size, SortingType_t So
     }
 }
 template<typename T>
-void SortingAlgrothim<T>::CountingSort(T arr[], uint32_t size,uint32_t Range, SortingType_t SortingType )
-{   Range++;
-    T Count_Arr[Range]={0};
+void SortingAlgrothim<T>::CountingSort(T arr[], uint32_t size, SortingType_t SortingType )
+{  
+    uint32_t Max_Element = arr[SortingAlgrothim::GetMaxIndex(arr,0,size-1)];
+    uint32_t Min_Element = arr[SortingAlgrothim::GetMinIndex(arr,0,size)];
+    uint32_t Range=Max_Element-Min_Element +1;
+    std::vector<T> Count_Arr(Range,0);
```

Comment:
@AbdelrahmanBadr22 
We are forced to make a vector like this to use index in implementaion , Ok?
Now , Whats difference between vector and dynamic array in this case if i do dynamic array like this : 
T * Count_Arr = new T (Range);

Reply:
That's a really good question. Using dynamic array such as `T * Count_Arr = new T (Range);` will be better in this case and for all cases that array size is not known at runtime and also not changed. 
`vector` will be better in case of  array size is not known at runtime and size is changed during the program.

so I think using dynamic array will be better as you mentioned. 

NOTE : **We are forced to make a vector like this to use index in implementaion , Ok?** 
Actually in vector implementation there are operator overloading for `[ ]` operator 

```
int main() 
{   vector<int>vec;
    int n,x;
    cout<<"Enter the size of the vector: ";
    cin>>n;
    cout<<"Enter vector values:\n";
    for (int i=0;i<n;i++)
    {
        cin>>x;
        vec.push_back(x);
    }
    cout<<"\nReference operator vec[3]= "<<vec[3]; 
```

Reply:
Great, I will do dynamic array then push it again and you are right about vector.
Thanks For You .